app/controllers/notifications_controller.rb

```diff
@@ -90,7 +90,19 @@ def mark_read
     if params[:id]
       Notification.read(current_user, [params[:id].to_i])
     else
-      Notification.where(user_id: current_user.id).includes(:topic).where(read: false).update_all(read: true)
+      if types = params[:dismiss_types]&.split(",").presence
+        types.map! do |type|
+          Notification.types[type.to_sym] || (
+            raise Discourse::InvalidParameters.new("invalid notification type: #{type}")
+          )
+        end
+      end
+      query = Notification
+        .where(user_id: current_user.id, read: false)
+        .includes(:topic)
+
+      query = query.where(notification_type: types) if types
+      query.update_all(read: true)
```

Comment:
Is there a way to extend `Notifcation.read` instead?

Reply:
I went with a new method `Notification.read_types` because `Notification.read` expects an array of notification IDs that should be marked as read whereas I need to mark all notifications of specific types as read. We also have `Notification.mark_posts_read` that does a very similar thing so I felt it makes sense to introduce a new method for marking notification types as read.